src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java

```diff
@@ -25,23 +46,44 @@ public record Error(String message) {
 
   }
 
-  private static final Duration SCHEMA_FETCH_ERROR_TTL = Duration.ofSeconds(30);
+  @ConfigProperty(name = "ide-sidecar.schema-fetch-error-ttl")
+  long schemaFetchErrorTtl;
 
   public static final Map<ConnectionId, Cache<SchemaId, SchemaErrors.Error>> cacheOfCaches = new ConcurrentHashMap<>();
 
-  public static Cache<SchemaId, Error> getSubCache(ConnectionId key) {
+  /**
+   * Retrieves or creates a sub-cache for a specific connection.
+   *
+   * @param key The connection identifier.
+   * @return The sub-cache for the connection.
+   */
+  public Cache<SchemaId, Error> getSubCache(ConnectionId key) {
     return cacheOfCaches.computeIfAbsent(
         key,
-        k -> Caffeine.newBuilder().expireAfterAccess(SCHEMA_FETCH_ERROR_TTL).build());
+        k -> Caffeine.newBuilder().expireAfterAccess(Duration.ofSeconds(schemaFetchErrorTtl)).build());
   }
 
+  /**
+   * Reads a schema error from the cache.
+   *
+   * @param connectionId The connection identifier.
+   * @param schemaId     The schema identifier.
+   * @return The schema error, or null if not found.
+   */
   public Error readSchemaIdByConnectionId(
       ConnectionId connectionId,
       SchemaId schemaId
   ) {
     return getSubCache(connectionId).getIfPresent(schemaId);
   }
```

Comment:
The record types primarily exist to simplify the definition of the map of maps, so we can be don't need to expose them outside the `SchemaErrors` class. Having these methods _require_ the record types makes these methods harder to use. So we could either:
1. Refactor the public methods to not use the record types; or
2. Add additional methods that don't use record types

For example, option 2 would add methods like:
```
  public Error readSchemaIdByConnectionId(String connectionId, String clusterId, String schemaId) {
    return readSchemaIdByConnectionId(
        new ConnectionId(connectionId),
        new SchemaId(clusterId, schemaId)
    );
  }
```

Reply:
I might be missing something... wouldn't this recurse as written?

Reply:
No, the `readSchemaIdByConnectionId(String, String, String)` method can call the `readSchemaIdByConnectionId(ConnectionId, SchemaId)` method and it will not recurse since they're different signatures.